src/main/java/com/epam/petclinic/order/config/OAuth2FeignAutoConfiguration.java

```diff
@@ -0,0 +1,22 @@
+package com.epam.petclinic.order.config;
+
+import feign.Feign;
+import feign.RequestInterceptor;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.oauth2.client.OAuth2ClientContext;
+
+@Configuration
+@ConditionalOnClass({ Feign.class })
```

Comment:
If my understanding is correct we will get compile exception if there is no Feign class in classpath (we need in to compile this class according to import clause)?